index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
I'm fine with leaving the results of invalid operations unspecified. This seems no different from a `map` function invoking the argument that is *meant* to be a function without first checking that it is in fact a function.